### PR TITLE
Update material models

### PR DESCRIPTION
**Pull request by allenwq**
Using gem edge

Add uniqueness constraint on name

### CODE REVIEW

File: app/models/course/material/folder.rb [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
-                        foreign_key: :parent_folder_id, class_name: Course::Material::Folder.name
+  acts_as_forest order: 'name'
+
+  after_initialize :set_defaults, if: :new_record?
+
   has_many :materials, inverse_of: :folder, dependent: :destroy, foreign_key: :folder_id,
                        class_name: Course::Material.name
+  belongs_to :course, inverse_of: :material_folders
+
+  validates :name, uniqueness: { case_sensitive: false, scope: :parent_id }, if: :parent
```
Comment (lowjoel) [inline app/models/course/material/folder.rb]: This should be done at the database, schema_plus_validations should be able to infer this.

Comment (allenwq) [inline app/models/course/material/folder.rb]: Yah, but It failed to infer the case_sensitive case, that's why I put it here.

Comment (lowjoel) [inline app/models/course/material/folder.rb]: Can you try to get it to work? I remembered fixing it before. Check the gem.

Comment (allenwq) [inline app/models/course/material/folder.rb]: OK, I'll check.  But to me the case_insensitive option is not standard rails but an extension of `schema_plus_pg_indexes`

Comment (lowjoel) [inline app/models/course/material/folder.rb]: yes, it is.

Comment (allenwq) [inline app/models/course/material/folder.rb]: Sure

**Comment by allenwq** [comment]
There is an issue in schema_validations, which says that it cannot infer the case-insensitive case for now:
https://github.com/SchemaPlus/schema_validations/issues/21

**Comment by lowjoel** [comment]
Can you place a comment in the code? We can remove it all when the gem starts doing inference.

**Comment by allenwq** [comment]
OK, Done !

**Pull request merged**